sdk/eventhub/event-hubs/src/sender.ts

```diff
@@ -154,7 +161,40 @@ export class EventHubProducer {
     if (!Array.isArray(eventData) && !(eventData instanceof EventDataBatch)) {
       eventData = [eventData];
     }
-    return this._eventHubSender!.send(eventData, { ...this._senderOptions, ...options });
+
+    // link message span contexts
+    let spanContextsToLink: SpanContext[] = [];
+    if (Array.isArray(eventData)) {
+      for (let i = 0; i < eventData.length; i++) {
+        const event: EventData = {...eventData[i], properties: {...eventData[i].properties}};
+        const messageSpan = createMessageSpan(options.parentSpan);
+        // since these message spans are created from same context as the send span,
+        // these message spans don't need to be linked.
+        instrumentEventData(event, messageSpan);
+        messageSpan.end();
+      }
+    } else if (eventData instanceof EventDataBatch) {
```

Comment:
instanceof checks can be a little brittle. I prefer checking for a property that is only on the specific type and creating a user defined type guard: https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

Reply:
Sure, I'll update to check the methods on the class. That'll make it easier to mock as well!

Reply:
Mental note, I still need to make this change.